sycl/test/usm/allocator_vector.cpp

```diff
@@ -25,32 +25,33 @@ int main() {
   auto dev = q.get_device();
   auto ctxt = q.get_context();
 
-  usm_allocator<int, usm::alloc::host> alloc(ctxt, dev);
+  if (dev.get_info<info::device::usm_host_allocations>()) {
```

Comment:
Instead of wrapping test code with `if (device supports USM) { ... }`, it's better to use early exit.
```c++
if (!dev.get_info<info::device::usm_host_allocations>())
  return 0;
... // tests using USM
```